### PR TITLE
[pkg/translator/signalfx] Extract signalfx to metrics conversion in a separate package

### PR DESCRIPTION
**Pull request by bogdandrutu**
Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

### CODE REVIEW

File: exporter/prometheusexporter/go.mod [hunk 1]
```diff
@@ -33,7 +33,6 @@ require (
 	github.com/aws/aws-sdk-go v1.42.48 // indirect
 	github.com/beorn7/perks v1.0.1 // indirect
 	github.com/cenkalti/backoff/v4 v4.1.2 // indirect
-	github.com/census-instrumentation/opencensus-proto v0.3.0 // indirect
```
Comment (dmitryax) [inline exporter/prometheusexporter/go.mod:36]: How these appeared in the changeset?

**Review by dmitryax — COMMENTED** [review]
Why do we need to extract it?

File: exporter/prometheusexporter/go.mod (hunk 1 above)
Comment (bogdandrutu) [reply exporter/prometheusexporter/go.mod:36]: I've ran `make gotidy`... not sure what changes happen :))

**Comment by bogdandrutu** [comment]
@dmitryax 

> Why do we need to extract it?

We want to use this library in other places (some internal) to convert to/from otlp to signalfx.

**Review by dmitryax — APPROVED** [review]

**Pull request merged**